### ISSUE TITLE
improve worker construction interface

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#2177]
Change the `*Worker` interfaces to resemble the interfaces in #2130.

The inputs to a worker should be everything that is necessary to run a job. Right now some of the main inputs (like images) are nested within helper classes like `IntegrationLauncher` which are not easily to enforce as JSON serializable. They also contain things like process builder factories, which are really options at the worker runner level more than the worker itself.

If the images are top-level arguments for workers this becomes much easier, but requires some refactoring where they are called.

To put it another way: make `WorkerRun` be the level of worker abstraction, not `Worker`

**Comment by cgardens** [comment]
PR up. I think we are aligned on the approach. Need to fix all the tests tomorrow.
